### PR TITLE
Fix off-by-one bug when searching lists

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

When filing another ticket, I noticed that we have an off-by-one error when searching lists - we only show the "nothing found" message after the user types the second character that results in a 0 item filtered list. Thanks to @MattLichtenstein who taught me some QML, and found that the fix was because it wasn't properly bound.

Relatedly, `_searchBarHasError` had recently changed from a callback to a boolean, but hadn't been appropriately updated in `VPNServerList.qml`. This PR also fixes that. (Hat tip to @MattLichtenstein for noticing this one when we were working together.)

## Reference

    N/A

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

**Review by MattLichtenstein — APPROVED** [review]
👌

**Comment by mcleinman** [comment]
> can you add a test?

Test has been added

**Review by mozrokafor — APPROVED** [review]

File: tests/functional/testServers.js [hunk 1]
```diff
@@ -238,6 +238,29 @@ describe('Server list', function() {
...
+    await vpn.setElementProperty(homeScreen.serverListView.SEARCH_BAR_TEXT_FIELD, 'text', 's', 'Austraz');
+    // ensure message is visible
+    await vpn.waitForElementProperty(homeScreen.serverListView.SEARCH_BAR_ERROR, 'visible', 'true');
+    // add another character
+    await vpn.setElementProperty(homeScreen.serverListView.SEARCH_BAR_TEXT_FIELD, 'text', 's', 'Austrazz');
+    // ensure message
+    await vpn.waitForElementProperty(homeScreen.serverListView.SEARCH_BAR_ERROR, 'visible', 'true');
+    // delete a couple characters
+    await vpn.setElementProperty(homeScreen.serverListView.SEARCH_BAR_TEXT_FIELD, 'text', 's', 'Austra');
+    // ensure message disappears
```
Comment (MattLichtenstein) [inline tests/functional/testServers.js:260]: Love the extra comments!

**Review by MattLichtenstein — APPROVED** [review]
Nice!

**Pull request merged**